src/lib/core/style/add-focus-classes.ts

```diff
@@ -0,0 +1,80 @@
+import {Directive, Injectable, Optional, SkipSelf} from '@angular/core';
+
+
+/** Singleton that allows all instances of CdkAddFocusClasses to share document event listeners. */
+@Injectable()
+export class CdkFocusCauseDetector {
```

Comment:
How about `FocusOriginMonitor`? I would like to avoid prefixing any classes with "cdk".